module/modmanager/manager_test.go

```diff
@@ -286,3 +280,152 @@ func TestModManagerValidation(t *testing.T) {
 	err = mgr.Close(ctx)
 	test.That(t, err, test.ShouldBeNil)
 }
+
+func TestModuleReloading(t *testing.T) {
+	ctx := context.Background()
+
+	myHelperModel := resource.NewModel("rdk", "test", "helper")
+	rNameMyHelper := generic.Named("myhelper")
+	cfgMyHelper := resource.Config{
+		Name:  "myhelper",
+		API:   generic.API,
+		Model: myHelperModel,
+	}
+	_, err := cfgMyHelper.Validate("test", resource.APITypeComponentName)
+	test.That(t, err, test.ShouldBeNil)
+
+	// This cannot use t.TempDir() as the path it gives on MacOS exceeds module.MaxSocketAddressLength.
+	parentAddr, err := os.MkdirTemp("", "viam-test-*")
```

Comment:
maybe we should single source this to a testutil?

Reply:
I tried to move this to a testutil, but it doesn't reduce code that much. The `defer` still needs to be called, so we only save a couple lines. I can still move it if you feel strongly, but I find this to be slightly more readable over the three occurrences.

Reply:
that's fine!